modules/hanna-react/src/Tooltip.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { CSSProperties, useRef } from 'react';
+import {
+  arrow as arrowPlugin,
+  autoUpdate,
+  flip,
+  offset,
+  Placement,
+  shift,
+  Side,
+  useFloating,
+} from '@floating-ui/react';
+import { useCallbackOnEsc } from '@hugsmidjan/react/hooks';
+import useLaggyState from '@hugsmidjan/react/hooks/useLaggyState';
+import getBemClass from '@hugsmidjan/react/utils/getBemClass';
+
+type TooltipElement = HTMLDetailsElement & {
+  $contextClicked_firefox_fix?: ReturnType<typeof setTimeout>;
+  $hasFocus?: ReturnType<typeof setTimeout>;
+};
+
+type SideFromPlacement<place = Placement> = place extends `${infer side}-${string}`
+  ? side
+  : never;
+
+const getSide = (placement: Placement): Side =>
+  placement.split('-')[0] as SideFromPlacement;
```

Comment:
```suggestion
const getSide = <P extends Placement>(placement: P) =>
  placement.split('-')[0] as SideFromPlacement<P>;
```

Minor atriði en líklega sniðugt að bæta við.